libs/vkd3d/device.c

```diff
@@ -4987,6 +4995,76 @@ struct d3d12_device *unsafe_impl_from_ID3D12Device(d3d12_device_iface *iface)
     return impl_from_ID3D12Device(iface);
 }
 
+static inline struct d3d12_device *d3d12_device_from_ID3D12DeviceExt(ID3D12DeviceExt *iface)
+{
+    struct d3d12_device_ext* d12device_ext = CONTAINING_RECORD(iface, struct d3d12_device_ext, ID3D12DeviceExt_iface);
+    return CONTAINING_RECORD(d12device_ext, struct d3d12_device, device_ext);
+}
+
+static ULONG STDMETHODCALLTYPE d3d12_device_ext_AddRef(ID3D12DeviceExt *iface)
+{
+    struct d3d12_device *device = d3d12_device_from_ID3D12DeviceExt(iface);
+    return ID3D12Device_AddRef(&device->ID3D12Device_iface);
+}
+
+static ULONG STDMETHODCALLTYPE d3d12_device_ext_Release(ID3D12DeviceExt *iface)
+{
+    struct d3d12_device *device = d3d12_device_from_ID3D12DeviceExt(iface);
+    return ID3D12Device_Release(&device->ID3D12Device_iface);
+}
+
+static HRESULT STDMETHODCALLTYPE d3d12_device_ext_QueryInterface(ID3D12DeviceExt *iface,
+        REFIID iid, void **out)
+{
+    TRACE("iface %p, iid %s, out %p.\n", iface, debugstr_guid(iid), out);
+
+    if (IsEqualGUID(iid, &IID_ID3D12DeviceExt) ||
```

Comment:
Again, I think we need to be able to query original interfaces from this interface. Perhaps we could just chain QueryInterface to the ID3D12Device impl